scripts/update_repos_version.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+import os
+import os.path
+import re
+import logging
+import subprocess
+import fileinput
+import shutil
+import click
+import github
+
+LOG_FILENAME = 'bump_repo_version_log.txt'
+logging.basicConfig(filename=LOG_FILENAME,level=logging.INFO)
+
+
+REPOS_TO_CHANGE = (
+    'https://github.com/edx/credentials.git',
+    'https://github.com/edx/course-discovery.git',
+    'https://github.com/edx/ecommerce.git',
+    'https://github.com/edx/programs.git',
+    'https://github.com/edx/edx-analytics-data-api.git',
+    'https://github.com/edx/edx-analytics-dashboard.git',
+    'https://github.com/edx/xqueue.git',
+    'https://github.com/edx/edx-platform.git',
+)
+
+# Here's where we'll clone the repos.
+TEMP_DIRECTORY = 'git_tmp'
+
+RELEASE_UTIL_REPO_NAME = 'edx-django-release-util'
+
+
+class GitHubApiUtils(object):
+    """
+    Class to query/set GitHub info.
+    """
+    def __init__(self, repo_id):
+        """
+        Returns a GitHub object, possibly authed as a user.
+        """
+        token = os.environ.get('GITHUB_TOKEN', '')
+        username = os.environ.get('GITHUB_USERNAME', '')
+        password = os.environ.get('GITHUB_PASSWORD', '')
+        if len(token):
+            self.gh = github.Github(login_or_token=token)
+        elif len(username) and len(password):
+            self.gh = github.Github(login_or_token=username, password=password)
+        else:
+            # No auth available - use the API anonymously.
+            self.gh = github.Github()
+        self.repo = self.gh.get_repo(repo_id)
+
+    def create_pull(self, *args, **kwargs):
+        return self.repo.create_pull(*args, **kwargs)
+
+
+@click.command()
+@click.option("--new_version", help="New version of edx-django-release-util.", type=str, required=True)
+def bump_repos_version(new_version):
+    """
+    Changes the pinned version number in the requirements files of all repos
+    which have edx-django-release-util as a dependency.
+
+    This script assumes the following environment variables are set for GitHub authentication:
+    Either: GITHUB_TOKEN -or- GITHUB_USERNAME & GITHUB_PASSWORD.
+    """
+    # Make the cloning directory and change directories into it.
+    original_dir = os.getcwd()
+    if not os.path.exists(TEMP_DIRECTORY):
+        os.makedirs(TEMP_DIRECTORY)
+    os.chdir(TEMP_DIRECTORY)
+    tmp_dir = os.getcwd()
+
+    # Iterate through each repository.
+    for repo in REPOS_TO_CHANGE:
+        repo_pattern = re.compile(r'^https://github.com/edx/(.+).git$')
+        repo_name = repo_pattern.match(repo).groups()[0]
```

Comment:
I'm going to remove this - and change the REPOS_TO_CHANGE above. Easier to build a URL than to parse out `user/repo`.